### ISSUE TITLE
VM snapshot should use a copy of virtual state metadata

### ISSUE THREAD

**Issue by artemananiev** [hiero-ledger/hiero-consensus-node#20468]
During reconnects, virtual map copy is created on the teacher side to send data to the learner. This is done in `VirtualMap.detach()` method:

```java
        final VirtualDataSource dataSourceCopy = dataSourceBuilder.copy(dataSource, false, false);
        final VirtualNodeCache cacheSnapshot = cache.snapshot();
        return new RecordAccessor(state, cacheSnapshot, dataSourceCopy);
```

This method creates a new record accessor using a cache snapshot and a data source copy. However, VM metadata (the "state" field in the code snippet above) is used as is. If it's mutated later, this record accessor may be broken. Instead, metadata copy should be used, similar to cache and data source copies.
